### PR TITLE
Fix init timing

### PR DESCRIPTION
**Pull request by fmauch**
This MR aims at fixing startup timing issues like
 * Too much time goes by between starting the pipeline and starting robot control
 * Too much time goes by between the RTDE handshake and starting the pipeline

Also, we added more error handling (of cases that usually should not occur) to the RTDE handshake process.

This should solve #6

### CODE REVIEW

File: ur_robot_driver/src/rtde/rtde_client.cpp [hunk 1]
```diff
@@ -47,14 +47,38 @@ RTDEClient::RTDEClient(std::string robot_ip, comm::INotifier& notifier, const st
...
+  pipeline_.init();
   pipeline_.run();
+
+  uint16_t protocol_version = negotiateProtocolVersion();
+  parser_.setProtocolVersion(protocol_version);
+
+  queryURControlVersion();
+  if (urcontrol_version_.major < 5)
+  {
+    max_frequency_ = CB3_MAX_FREQUENCY;
```
Comment (gavanderhoorn) [inline ur_robot_driver/src/rtde/rtde_client.cpp]: Could be nice to notify users of this. Or at least add a `ROS_DEBUG..(..)`.

Comment (fmauch) [reply ur_robot_driver/src/rtde/rtde_client.cpp]: It actually is still there. However, it got refactored into the `setOutputs()` function which is called directly afterwards.

Do you think, that would be nice to do, nonetheless? Output would then be something like
```
[ DEBUG] [1572871771.209652246]: Setting RTDE communication frequency to 125 Hz, as using a CB3 robot.
[ INFO] [1572871771.209652246]: Setting up RTDE communication with frequency 500.000000
```

**Comment by fmauch** [comment]
@t-schnell please review

File: ur_robot_driver/include/ur_robot_driver/comm/producer.h [hunk 2]
```diff
@@ -90,8 +99,10 @@ class URProducer : public IProducer<HeaderT>
    *
    * \returns Success of reading and parsing the package
    */
-  bool tryGet(std::vector<std::unique_ptr<URPackage<HeaderT>>>& products)
+  bool tryGet(std::vector<std::unique_ptr<URPackage<HeaderT>>>& products) override
   {
+    // TODO This function has become really ugly! That should be refactored!
+
```
Comment (t-schnell) [inline ur_robot_driver/include/ur_robot_driver/comm/producer.h]: Thinking about it more I don't think it has actually become worse, but some refactoring (or at least adding some comments) still seems like a good idea.

File: ur_robot_driver/include/ur_robot_driver/comm/pipeline.h [hunk 3]
```diff
@@ -389,6 +401,8 @@ class Pipeline
     {
       if (!producer_.tryGet(products))
       {
+        producer_.teardownProducer();
+        running_ = false;
         break;
       }
```
[the remark this answers is not in the document]
Comment (fmauch) [reply ur_robot_driver/include/ur_robot_driver/comm/pipeline.h]: I think, tearing it down is actually reasonable here, as if this occurs, we want to reconnect anyway.

File: ur_robot_driver/include/ur_robot_driver/comm/producer.h (hunk 2 above)
Comment (fmauch) [reply ur_robot_driver/include/ur_robot_driver/comm/producer.h]: Yes, it was worse in between... But I'd like to leave this comment there for the moment.

**Pull request merged**